src/java.base/share/classes/java/lang/reflect/code/op/CoreOp.java

```diff
@@ -1953,14 +1942,17 @@ public ArrayLoadOp transform(CopyContext cc, OpTransformer ot) {
             }
 
             ArrayLoadOp(Value array, Value index) {
+                this(array, index, ((ArrayType)array.type()).componentType());
```

Comment:
Can you make a note to revisit this when the component type is not explicitly given (see `VarOp.resultType` as an example). The current approach allows for easier transformation of the type because it is a function of inputs (the operand type). I know we are not consistent in this area and need a better general solution.